f-string-without-interpolation false positive with t-strings when using format spec

### Bug description

```python
When passing any format specs to t-strings, an f-string-without-interpolation warning will be filed.

Consider the following snippet:

def f():
    pi = 3.14
    return t"{pi:.1f}"


The template string line will be flagged by lint as having the f-string-without-interpolation issue.
```

### Configuration

```ini

```

### Command used

```shell
python -m pylint test.py
```

### Pylint output

```python
************* Module test
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:3:16: W1309: Using an f-string that does not have any interpolated variables (f-string-without-interpolation)
```

### Expected behavior

Expected behavior is for no linting error to be raised on format spec provided with template string.

### Pylint version

```shell
pylint 4.0.2
astroid 4.0.1
Python 3.14.0 (main, Oct 19 2025, 09:04:15) [GCC 13.3.0]
```

### OS / Environment

WSL 2: Ubuntu 24.04.1 LTS

### Additional dependencies

```python

```

--- a/doc/whatsnew/fragments/10702.false_positive
+++ b/doc/whatsnew/fragments/10702.false_positive
@@ -1,0 +1,4 @@
+Fix false positive for ``f-string-without-interpolation`` with template strings
+when using format spec.
+
+Closes #10702

--- a/pylint/checkers/strings.py
+++ b/pylint/checkers/strings.py
@@ -409,7 +409,7 @@ class StringFormatChecker(BaseChecker):
         self._check_interpolation(node)
 
     def _check_interpolation(self, node: nodes.JoinedStr) -> None:
-        if isinstance(node.parent, nodes.FormattedValue):
+        if isinstance(node.parent, (nodes.TemplateStr, nodes.FormattedValue)):
             return
         for value in node.values:
             if isinstance(value, nodes.FormattedValue):

--- a/tests/functional/f/f_string_without_interpolation_py314.py
+++ b/tests/functional/f/f_string_without_interpolation_py314.py
@@ -1,0 +1,4 @@
+# pylint: disable=missing-module-docstring
+
+PI = 3.14
+print(t"{PI:.1f}")

--- a/tests/functional/f/f_string_without_interpolation_py314.rc
+++ b/tests/functional/f/f_string_without_interpolation_py314.rc
@@ -1,0 +1,2 @@
+[testoptions]
+min_pyver=3.14

